lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -205,7 +206,7 @@ def store_content(resource)
       Puppet.info "Content for '#{resource[:source]}' already exists"
     else
       Puppet.info "Storing content for source '#{resource[:source]}'"
-      content = Puppet::FileServing::Content.indirection.find(resource[:source])
+      content = Puppet::FileServing::Content.indirection.find(resource[:source], {:environment => request.environment})
```

Comment:
@parkerd Thanks for the offer. I think it's actually ok, because the `store_content` method is only called if we were able to retrieve metadata for the file, so we had to at least stat'ed the file. There is a race condition where the file could be deleted after we retrieve it's metadata and before we attempt to read its content, but it's unlikely.

I'll take a look at adding a test to ensure the correct environment is threaded through to the FileMetadata and FileContent methods, and merge this.